### ISSUE TITLE
Editor: Desktop app when the WordPress.com editor is unavailable

### ISSUE THREAD

**Issue by creativecoder** [Automattic/wp-calypso#41417]
Investigate the impact of the WordPress.com editor deprecation on the Desktop app

Blocked by #41087 and #41108

## Requirements

1. Test the Desktop app when the WordPress.com editor is not available by toggling between the Block and Classic editors.
1. Create additional issues for any problems that need to be fixed that prevent users from toggling between editors and editing content.

## Notes

From pb5gDS-pT-p2

> I’d also recommend checking to see how the Desktop App responds when we don’t have the Calypso Classic editor available.

**Comment by gwwar** [comment]
It looks like @nsakaimbo might have spent some time looking at fallback behavior as well, in case y'all haven't talked yet.

**Comment by nsakaimbo** [comment]
@gwwar @creativecoder Hello! I'm actually in the process of debugging an editor bug on the desktop app right now.

**TL;DR**: there are cases where clicking "new post" routes the user to WP Admin in a browser, instead of staying within the Desktop app/Calypso.

https://github.com/Automattic/wp-desktop/issues/640

It looks like this error has reared it's head in various forms over time, but it's occuring for several users currently and I'm also able to reproduce this 100% of the time.

There's also this currently-open issue, where the first load of an existing draft re-directs to WP-Admin: #40376

I've been trying to understand the editor/authentication logic a little bit more but would love to pair on this, if anyone is available/willing to help dig into this further.

**Comment by creativecoder** [comment]
Thanks for the information @nsakaimbo!

> I've been trying to understand the editor/authentication logic a little bit more but would love to pair on this, if anyone is available/willing to help dig into this further.

We're not actively working on this issue yet, but will be in the next week or so--I'll make sure we check in with you when we start it.

**Comment by creativecoder** [comment]
Noting from https://github.com/Automattic/wp-desktop/issues/640#issuecomment-622535806 that Jetpack SSO has to be enabled for a user to be able to use the block editor from Calypso or the classic editor in wp-admin

**Comment by pento** [comment]
@nsakaimbo: How's the exploration going with messaging around Jetpack SSO?

I've been thinking about this problem with regards to Simple Sites who have opted to use the wp-admin classic editor: it seems like there would be a substantial amount of work involved in getting the classic editor working inside the Calypso iframe, which is difficult to justify when all new users (and all development work) is focussed on the block editor.

I'm wondering if the logic here could be contained within the iframe component? As you've noted with the several bugs this causes, the current behaviour of redirecting to the site is pretty bad. Could we simplify this down?

- If it's a Jetpack site that doesn't have SSO enabled, show a message that they need to enable it.
- If it's any site that's trying to load the wp-admin classic editor, show a message that this editor isn't supported in the desktop app.

In both cases, offer to open the editor in an external browser, instead.

**Comment by nsakaimbo** [comment]
@pento Thanks for the checking in, and for the suggestion.

I agree - at the very least additional messaging is necessary to help unblock users (as well as minimize confusion).

I have some basic messaging/UI ready-to-go, except I'm struggling with a couple of technical issues:

* _Warn Jetpack users trying to user Gutenberg_: Within Calypso, both Simple and Jetpack sites use some redirection before the Gutenberg editor is ready. Therefore, checking for a redirect event isn't a surefire way to indicate that the action is being attempted on a Jetpack site. **TL;DR**: how can the Desktop app distinguish between a Simple site and a Jetpack site, in order to display the necessary UI?

* _Warn Simple users trying to use Classic_: This has not been on my radar yet (been focused on the above), but now that you mention it, it totally should be. In this case, I'm unclear on where the Desktop app can hook into to know that the user is attempting to load the Classic editor.

If both of the above are fairly straightforward w.r.t the code paths in Calypso (although I don't know where at the moment because of lack of familiarity with the codebase), perhaps we can add a couple of redux events the Desktop app can subscribe to? (Perhaps something like `JETPACK_SITE_SELECTED` and `LOAD_CLASSIC_EDITOR`?)

What do you think?

**Comment by pento** [comment]
I'm closing this issue as fixed, it appears to have all been addressed by @nsakaimbo's work. 🙂
